lib/features/mailbox/presentation/mixin/mailbox_widget_mixin.dart

```diff
@@ -248,6 +274,23 @@ mixin MailboxWidgetMixin {
       .toList();
   }
 
+  bool isSubaddressingSupported(Session? session, AccountId? accountId) {
+    if (session == null || accountId == null) {
+      return false;
+    }
+    if (!CapabilityIdentifier.jmapTeamMailboxes.isSupported(session, accountId)) {
+      return false;
+    }
+
+    final subaddressingSupported = ((session
+        .props[0] as Map<CapabilityIdentifier, CapabilityProperties>?)
+        ?[capabilityDeletedMessagesVault]
+        ?.props[0] as Map<String, dynamic>?)
+        ?['subaddressingSupported'];
+
+    return subaddressingSupported ?? true;
```

Comment:
this check is related to [this new ticket](https://github.com/linagora/tmail-backend/issues/1324): we only want the subaddressing options to be available if the server explicitly advertizes that it supports it.

since it does not for now, the check will always fail, so i leave the default value to `true` for testing purposes. this will have to be changed to `false` before merging